dig.go

```diff
@@ -314,6 +314,47 @@ type Container struct {
 
 	// invokerFn calls a function with arguments provided to Provide or Invoke.
 	invokerFn invokerFn
+
+	gh *graphHolder
+}
+
+// graphHolder represents the dependency graph for a Container. Specifically,
+// it saves constructorNodes and paramGroupedSlices (value groups) as graphNodes
+// and implements the Graph interface defined in internal/graph to run graph
+// algorithms on it. It has a 1-to-1 correspondence with a Container whose graph
+// it represents.
+type graphHolder struct {
+	// all the nodes defined in the graph.
+	allNodes []*graphNode
+
+	// Maps each graphNode to its order.
```

Comment:
High-level comment here and in-general about the documentation:
That "graph order" refers to the number of vertices is obvious,
but that "node order" refers to the index of the node in allNodes is not obvious.